rundeckapp/src/main/groovy/rundeck/services/audit/AuditEventsService.groovy

```diff
@@ -0,0 +1,534 @@
+        plugin.instance.init()
+
+        return new DescribedPlugin<AuditEventListener>(plugin.instance, pluginDescription, pluginDescription.name)
+    }
+
+    /**
+     * Handles and authentication success event
+     * @param event
+     */
+    @EventListener
+    void handleAuthenticationSuccessEvent(AuthenticationSuccessEvent event) {
+        eventBuilder()
+                .setUsername(extractUsername(event.authentication))
+                .setUserRoles(extractAuthorities(event.authentication))
+                .setActionType(ActionTypes.LOGIN_SUCCESS)
+                .setResourceType(ResourceTypes.USER)
+                .setResourceName(extractUsername(event.authentication))
+                .publish()
+    }
+
+    @EventListener
+    void handleAuthenticationFailureEvent(AuthenticationFailureBadCredentialsEvent event) {
+        if (!event.authentication) {
+            LOG.error("Null authentication on login failure event. Cancelling event dispatch.")
+            return
+        }
+
+
+        eventBuilder()
+                .setUsername(extractUsername(event.authentication))
+                .setUserRoles(extractAuthorities(event.authentication))
+                .setActionType(ActionTypes.LOGIN_FAILED)
+                .setResourceType(ResourceTypes.USER)
+                .setResourceName(extractUsername(event.authentication))
+                .publish()
+    }
+
+
+    /**
+     * Captures logout events.
+     *
+     */
+    @Override
+    void logout(HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
+        if (!authentication) {
+            LOG.error("Null authentication on logout event. Cancelling event dispatch.")
+            return
+        }
+
+        eventBuilder()
+                .setUsername(extractUsername(authentication))
+                .setUserRoles(extractAuthorities(authentication))
+                .setActionType(ActionTypes.LOGOUT)
+                .setResourceType(ResourceTypes.USER)
+                .setResourceName(extractUsername(authentication))
+                .publish()
+    }
+
+
+    /**
+     * Creates a new event builder.
+     * After the builder is configured, call {@link AuditEventBuilder#publish()} to publish the event.
+     * The builder can be published several times.
+     *
+     * @param eventType
+     * @return
+     */
+    AuditEventBuilder eventBuilder() {
+        return new AuditEventBuilder()
+    }
+
+
+    /**
+     * Add a new listener to the internal register.
+     * @param listener
+     */
+    void addListener(AuditEventListener listener) {
+        internalListeners.add(listener)
+    }
+
+    /**
+     * Removes the listener from the internal listener register.
+     * @param listener
+     */
+    void removeListener(AuditEventListener listener) {
+        internalListeners.remove(listener)
+    }
+
+
+    /**
+     * Dispatch the event.
+     */
+    private dispatchEvent(AuditEventBuilder eventBuilder) {
```

Comment:
grails has an internal eventing system, i think perhaps that should be used instead. the Interceptor can send an event that way, and the AuditEventsService can listen for it.  that way future events can be sent from original location without binding to this class